### PR TITLE
Added logger timestamp. 

### PR DESCRIPTION
**Pull request by pbattaglia**
This makes it easier to monitor queue durations.

### CODE REVIEW

File: weather_dl/weather-dl [hunk 1]
```diff
@@ -25,7 +25,14 @@ import tempfile
...
 if __name__ == '__main__':
-    logging.getLogger().setLevel(logging.INFO)
+    logger = logging.getLogger()
+    fmt = '%(levelname)s %(asctime)s %(name)s: %(message)s'
+    datefmt = '%y-%m-%d %H:%M:%S'
+    formatter = logging.Formatter(fmt=fmt, datefmt=datefmt)
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.root.addHandler(handler)
+    logger.setLevel(logging.INFO)
```
Comment (alxmrs) [inline weather_dl/weather-dl:35]: This should probably be moved to the `configure_logger` function. 

https://github.com/google/weather-tools/blob/ecb194c7bc62d9d0ad3982cce24eb745a470be84/weather_dl/download_pipeline/pipeline.py#L48

File: weather_dl/download_pipeline/parsers.py [hunk 2]
```diff
@@ -352,10 +353,12 @@ def prepare_target_name(config: Config) -> str:
...
     target_path = t.cast(str, parameters.get('target_path', ''))
     target_filename = t.cast(str, parameters.get('target_filename', ''))
-    partition_keys = t.cast(t.List[str], parameters.get('partition_keys', list()))
+    partition_keys = t.cast(t.List[str],
+                            cp.copy(parameters.get('partition_keys', list())))
 
     if use_date_as_directory(config):
-        date_vals = config['selection']['date'][0].split('-')
+        date = t.cast(str, config['selection']['date'][0])
+        date_vals = date.split('-')
```
Comment (alxmrs) [inline weather_dl/download_pipeline/parsers.py:361]: Please merge with main / clean up the commit history.

**Review by alxmrs — CHANGES_REQUESTED** [review]

**Review by alxmrs — APPROVED** [review]
LGTM. Would you mind pasting a preview of what the logs look like after this change (e.g. a screenshot or a codeblock)? Before I merge, I want to understand what the effect will be.

**Comment by pbattaglia** [comment]
Here's an example of what the logging looks like:

```INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.196Z: JOB_MESSAGE_DEBUG: Workflow config is missing a default resource spec.
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.218Z: JOB_MESSAGE_DEBUG: Adding StepResource setup and teardown to workflow graph.
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.241Z: JOB_MESSAGE_DEBUG: Adding workflow start and stop steps.
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.264Z: JOB_MESSAGE_DEBUG: Assigning stage ids.
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.374Z: JOB_MESSAGE_DEBUG: Executing wait step start13
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.429Z: JOB_MESSAGE_BASIC: Executing operation GroupBy request limits/GroupByKey/Create
INFO 22-04-05 12:30:33 apache_beam.runners.dataflow.dataflow_runner: 2022-04-05T11:30:32.469Z: JOB_MESSAGE_DEBUG: Starting worker pool setup.```

**Pull request merged**